Core/src/IfcGeometryConverter/SolidModelConverter.h

```diff
@@ -948,8 +953,25 @@ namespace OpenInfraPlatform
 				}
 
 				// direction and length of extrusion
-				const double depth = extrudedArea->Depth * this->UnitConvert()->getLengthInMeterFactor();
-				carve::geom::vector<3> extrusionVector = placementConverter->convertIfcDirection(extrudedArea->ExtrudedDirection);
+				const double depth = (typename IfcEntityTypesT::IfcLengthMeasure)(extrudedArea->Depth) * UnitConvert()->getLengthInMeterFactor();
+				carve::geom::vector<3>  extrusionVector = carve::geom::VECTOR(0.0, 0.0, 0.0);
+				auto& vecDirection = extrudedArea->ExtrudedDirection->DirectionRatios;
+				
+				switch (vecDirection.size()) 
+				{
+				case 3:
+				{
+					extrusionVector.z = vecDirection[2] * depth;
+				}
+				case 2:
+				{
+					extrusionVector.x = vecDirection[0] * depth;
+					extrusionVector.y = vecDirection[1] * depth;
+					break;
+				}
+				default:
+					throw oip::InconsistentModellingException(extrudedArea, "Cardinality of the direction ratios can be only 2 or 3");
+				}

```

Comment:
Please use placementconverter::convertIfcDirection

Reply:
I have used placementconverter::convertIfcDirection before, and it has caused exactly those changes to IFC files, that I was trying to revert. I guess I found a solution, how to correctly use placementconverter::convertIfcDirection and do not cause this problem anymore.

Reply:
Found the solution. Resolved: 392efce